core/src/main/java/org/apache/metamodel/data/SimpleDataSetHeader.java

```diff
@@ -104,12 +104,6 @@ public int indexOf(SelectItem item) {
             }
             i++;
         }
-        
-        final boolean scalarFunctionQueried = item.getScalarFunction() != null;
-        if (scalarFunctionQueried) {
-            final SelectItem itemWithoutFunction = item.replaceFunction(null);
-            return indexOf(itemWithoutFunction);
```

Comment:
Similarly, this special if-block for scalar functions was kind of a bad code smell that's now gone :-)